data/teca_calendar_util.cxx

```diff
@@ -783,6 +785,92 @@ int day_iterator::get_next_interval(time_point &first_step,
     return 0;
 }
 
+// --------------------------------------------------------------------------
+bool n_steps_iterator::is_valid() const
+{
+    if (!this->valid)
+        return false;
+
+    if (this->index + this->number_of_steps - 1 >= this->time->size())
+    {
+        return false;
+    }
+
+    return true;
+}
+
+int n_steps_iterator::set_number_of_steps(long n)
+{
+    this->number_of_steps = n;
+}
+
+// --------------------------------------------------------------------------
+int n_steps_iterator::initialize(const const_p_teca_variant_array &t,
+    const std::string &units, const std::string &calendar,
```

Comment:
calendar and units are not used in this function so cast to void to silence the compiler warning. eg `(void) calendar;`